### ISSUE TITLE
Component Action

### ISSUE THREAD

**Issue by freesuraj** [wangshengjia/LeeGo#62]
Hi Wang,

Congrats on the great tool, nicely done !! I was wondering what's your plan for providing actions for components, like Button, swipe, tap, or anything? Blockable actions?

**Comment by wangshengjia** [comment]
Thanks a lot for the kind words, really glad to hear. 😀

For the moment, LeeGo support such kind of `Action` as IB files. Which means you can find and link the instance of component directly with `lg_viewForOutletKey` method, just as `IBOutlet` do.

```swift
// Retrieve target view from hierarchy based on a configured key in `Brick`, 
// then either manipulate object directly or keep it into a class property with 
// weak reference, just as `IBOutlet`
let likeButton = tweetView.lg_viewForOutletKey("likeButton")
likeButton.addTarget(...)
```

Since this approach let people manipulate directly instance which could introduce easily more mutable states and complexity, which isn't encouraged by LeeGo's philosophy, so we may use declarative blockable actions to replace it in the future release.

What you think?

**Comment by freesuraj** [comment]
Exactly. I would think declarative blocks should be the way to go. I and my colleague would be happy to contribute some as we love the work and the concept you've done so far. I believe we could make this one of the best componentKit out there.

**Comment by wangshengjia** [comment]
Yeah, totally cool. It would be great if you'd like to contribute, PR is always welcomed. 👍
